### ISSUE TITLE
Fuse generators with motors

### ISSUE THREAD

**Issue by MisterCheeeez** [Eggs-D-Studios/wos-issues#640]
### Guidelines

- [X] I checked for duplicate suggestions
- [X] My suggestion intends to add or improve something

### Version

Main (Production)

### Topic

A game mechanic (e.g. a part, a feature, etc)

### Why are you making your suggestion?

Fusing motors and generators opens up the opportunity to make custom power generation systems, as well as restore generators.

### What do you propose? What should change, or what should be added?

If motors are fused with generators, it may be able to bring them back through the motor, [which were previously removed](https://wiki.eggsdstudios.com/wiki/Generator#:~:text=Due%20to%20being%20abused%20by%20many%20players%20to%20create%20infinite%20electricity%2C%20the%20generator%20has%20been%20removed%20from%20Waste%20of%20Space), as well as add the ability to make custom power systems, such as turbines for water, steam, hydrogen, etc. Currently, some gasses have no purpose, and this could introduce them to the power generation field. As well as this, this could bring the in game help menu up to date, as it still contains info on generators as if they were never removed.

**Comment by dogranger1** [comment]
Re-implementing ``Generator``s is very unlikely for the same reason that it was removed (i.e. it can be used to generate infinite electricity).

**Comment by MisterCheeeez** [comment]
how?

**Comment by dogranger1** [comment]
> How? I wasn't a player at the time it was removed.

``Generator``s produced energy at a higher rate than ``Motor``s would consume energy, meaning zero effort would be required to set up a reliable source of energy. This idea does however remove that possibility, though the power output of the ``Generator`` would have to be nerfed (making it pretty useless) and there are no actual standardised methods to making objects spin without a  power output - other than using ``Engine``s (which require ``Gasoline``, though ``Gasoline`` requires energy to be created in the first place). ``Generator``s got removed alongside ``WindTurbine``s, which were used to make them spin (for context as to why they were there in the first place). 

Using gasses to rotate a turbine is a good idea, though I'm fairly certain that the way gasses work currently (they only apply a force upwards and not outwards, they also have a fixed volume and velocity) prevents them from efficiently moving a rotating object. Then there's also the consideration of how much energy would be produced in comparison to how much effort would be required to set up a build / how much energy must go into producing the gas.

**Comment by MisterCheeeez** [comment]
Yeah, although, aren't gasses floaty enough to be pushed though, and if not, cant we use them straight up?

You could probably set up an extractor to a tank for helium, then drop it out with a faucet, and when it floats upwards, make that upwards force turn the turbine by sending it _directly_ up, and let it go out of a hole in the top. Only if this makes it in the game, though.

**Comment by dogranger1** [comment]
> Yeah, although, aren't gasses floaty enough to be pushed though, and if not, cant we use them straight up?
> 
> You could probably set up an extractor to a tank for helium, then drop it out with a faucet, and when it floats upwards, make that upwards force turn the turbine by sending it _directly_ up, and let it go out of a hole in the top. Only if this makes it in the game, though.

If you really wanted to, then its possible, though its still not a good option considering how little force is applied to the turbine from 1 ``Hydrogen`` (hence little rotation occurs, which results in the issue I mentioned in the second paragraph). Then there's also the possibility of the turbine getting stuck due to gasses having a fixed shape and volume, meaning an inconsistent force is applied to the turbine (potentially leading to it getting stuck in a half turn).

**Comment by MisterCheeeez** [comment]
wow i COMPLETELY forgot about this until searching to make a new issue for similar ones

**Comment by hail12pink** [comment]
generators could be fixed by making them need to be directly attached to hinge and just checking the thing rotating them. AND making them not work with motors because that makes zero sense. maybe wind turbines (which should be readded aswell) could be connected to drivebox or whatever and that would could on generator too.
